clippy_lints/src/map_err_ignore.rs

```diff
@@ -0,0 +1,117 @@
+use crate::utils::span_lint_and_sugg;
+use rustc_errors::Applicability;
+use rustc_hir::{CaptureBy, Expr, ExprKind, PatKind, QPath};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for instances of `map_err(|_| Some::Enum)`
+    ///
+    /// **Why is this bad?** This map_err throws away the original error rather than allowing the enum to bubble the original error
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// enum Errors {
+    ///    Ignore
+    ///}
+    ///fn main() -> Result<(), Errors> {
+    ///
+    ///    let x = u32::try_from(-123_i32);
+    ///
+    ///    println!("{:?}", x.map_err(|_| Errors::Ignore));
+    ///
+    ///    Ok(())
+    ///}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// enum Errors {
+    ///    WithContext(TryFromIntError)
+    ///}
+    ///fn main() -> Result<(), Errors> {
+    ///
+    ///    let x = u32::try_from(-123_i32);
+    ///
+    ///    println!("{:?}", x.map_err(|e| Errors::WithContext(e)));
+    ///
+    ///    Ok(())
+    ///}
```

Comment:
Lets update this to be as idiomatic as possible:

- instead of fn `main` make up something more specific that we can artificially produce an error for, here's an example from the `unwrap_in_result` lint that we might want to copy:

```rust
fn divisible_by_3(i_str: String) -> Result<(), String> {
    let i = i_str
        .parse::<i32>()
        .map_err(|e| format!("cannot divide the input by three: {}", e))?;

    if i % 3 != 0 {
        Err("Number is not divisible by 3")?
    }

    Ok(())
}
```

We'll want to update this to use a proper error enum, I'm thinking:

```rust
enum Error {
    Indivisible {
        source: TryFromIntError,
        input: String,
    },
}
```

Additionally we should implement `Display` and `Error` for this type in the example.

Reply:
I update this, please take a look at your earliest convenience 😄